integrationTests/chainSimulator/staking/jail_test.go

```diff
@@ -0,0 +1,145 @@
+func TestChainSimulator_ValidatorJailUnJail(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	t.Run("staking ph 4 is not active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 4, "new")
+	})
+
+	t.Run("staking ph 4 step 1 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 5, "auction")
+	})
+
+	t.Run("staking ph 4 step 2 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 6, "auction")
+	})
+
+	t.Run("staking ph 4 step 3 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 7, "auction")
+	})
+}
+
+func testChainSimulatorJailAndUnJail(t *testing.T, targetEpoch int32, nodeStatusAfterUnJail string) {
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	numOfShards := uint32(3)
+
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   false,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              numOfShards,
+		GenesisTimestamp:         startTime,
+		RoundDurationInMillis:    roundDurationInMillis,
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  1,
+		NumNodesWaitingListShard: 1,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.EpochConfig.EnableEpochs.StakingV4Step1EnableEpoch = 5
+			cfg.EpochConfig.EnableEpochs.StakingV4Step2EnableEpoch = 6
+			cfg.EpochConfig.EnableEpochs.StakingV4Step3EnableEpoch = 7
+
+			cfg.EpochConfig.EnableEpochs.MaxNodesChangeEnableEpoch[2].EpochEnable = 7
+
+			newNumNodes := cfg.SystemSCConfig.StakingSystemSCConfig.MaxNumberOfNodesForStake + 8 // 8 nodes until new nodes will be placed on queue
+			configs.SetMaxNumberOfNodesInConfigs(cfg, newNumNodes, numOfShards)
+
+			cfg.RatingsConfig.ShardChain.RatingSteps.ConsecutiveMissedBlocksPenalty = 100
+			cfg.RatingsConfig.ShardChain.RatingSteps.HoursToMaxRatingFromStartRating = 1
+			cfg.RatingsConfig.MetaChain.RatingSteps.ConsecutiveMissedBlocksPenalty = 100
+			cfg.RatingsConfig.MetaChain.RatingSteps.HoursToMaxRatingFromStartRating = 1
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+	defer func() {
+		_ = cs.Close()
+	}()
+
+	metachainNode := cs.GetNodeHandler(core.MetachainShardId)
+	err = cs.GenerateBlocks(30)
+	require.Nil(t, err)
+
+	_, blsKeys, err := chainSimulator.GenerateBlsPrivateKeys(1)
+	require.Nil(t, err)
+
+	mintValue := big.NewInt(0).Mul(oneEGLD, big.NewInt(3000))
+	walletKeyBech, err := cs.GenerateAndMintWalletAddress(core.AllShardId, mintValue)
+	require.Nil(t, err)
+
+	walletKey, err := metachainNode.GetCoreComponents().AddressPubKeyConverter().Decode(walletKeyBech)
+	require.Nil(t, err)
```

Comment:
Not necessary here, but for further development we could compress all these calls in a single func to generate owners with stake + bls keys

Reply:
refactored this method